server/src/main/java/org/zanata/magpie/api/dto/CredentialDto.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2017, Red Hat, Inc. and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.zanata.magpie.api.dto;
+
+import java.util.Arrays;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+/**
+ * @author Patrick Huang
+ * <a href="mailto:pahuang@redhat.com">pahuang@redhat.com</a>
+ */
+public class CredentialDto {
+    private String username;
+    private char[] secret;
+
+    public CredentialDto(String username, char[] secret) {
+        this.username = username;
+        this.secret = nullSafeArrayCopy(secret);
+    }
+
+    public CredentialDto() {
+    }
+
+    @NotNull
+    @Size(min = 2, max = 20)
```

Comment:
Why 20?

Reply:
Just some arbitrary number. Changed to 128